src/kagglehub/clients.py

```diff
@@ -294,7 +296,7 @@ def post(self, data: dict, handle_path: str, resource_handle: Optional[ResourceH
         with requests.post(
             url,
             data=json.dumps(data),
-            auth=self._get_http_basic_auth(),
+            auth=self._get_auth(),
```

Comment:
@rosbo I am not familiar with `ColabClient`. Is this a shortcut cache for user to save models on Colab VM? Do we have any documents about this? Thanks!

Reply:
@mayankmalik-colab from the Colab team built this resolver. They have a few models (Gemma) that are pre-loaded on Kaggle and this Colab resolver return the path from a local disk for this subset of models.

Here is the feature bug for context: 303240466

We have unit test with stubs to test this resolver: https://github.com/Kaggle/kagglehub/blob/main/tests/test_colab_cache_model_download.py

To test it fully e2e, open a Colab notebook (not on corp b/c corp doesn't have internet), then install kagglehub from this branch and you can test it. @mayankmalik-colab can you share the slug of a model that is currently pre-loaded on Colab?

```
!pip install --force-reinstall --no-deps git+https://github.com/Kaggle/kagglehub.git@limagoog/feat/colab_secrets
``` 

The reason why the colab resolver needs the Kaggle credentials is that models such as Gemma requires user consent. The colab runtime calls Kaggle to check if the user has consented before granting them access to the pre-cached model.

Reply:
Sure, you can refer [this notebook](https://colab.research.google.com/drive/1okbquY1FrjJgWVoOg1SlvHAFD9QiRpsd?usp=sharing) and run on a GPU runtime.